### PR TITLE
Improve error message on non Win

### PR DESCRIPTION
**Pull request by CristiFati**
#422 in a different branch.

### CODE REVIEW

**Comment by CristiFati** [comment]
In the 1st commit I tried to keep things *Python 2* compatible as well (that's why I didn't use `raise from` from the beginning).

File: comtypes/__init__.py [hunk 1]
```diff
@@ -4,7 +4,16 @@
...
-from _ctypes import COMError
+try:
+    from _ctypes import COMError
+except ImportError as e:
+    msg = "\n".join((
+        "",
+        "COM technology not available (maybe it's the wrong platform).",
+        "Note that COM is only supported on Windows.",
+        "For more details, please check: "
+        "https://learn.microsoft.com/en-us/windows/win32/com."))
```
Comment (junkmd) [inline comtypes/__init__.py:15]: The final `.` in this line is not necessary.
The dots in the URL and the dots as punctuation are confusing.

File: comtypes/__init__.py [hunk 2]
```diff
@@ -4,7 +4,16 @@
 import atexit
 from ctypes import *
 from ctypes import _SimpleCData
-from _ctypes import COMError
+try:
+    from _ctypes import COMError
+except ImportError as e:
+    msg = "\n".join((
+        "",
```
Comment (junkmd) [inline comtypes/__init__.py:11]: Is this first blank line necessary?

**Review by junkmd — CHANGES_REQUESTED** [review]

**Comment by CristiFati** [comment]
I placed the empty line to be (what I consider) clearer. Of course it can be removed. Below are the differences (w / wo it):

```
[cfati@cfati-5510-0:/mnt/e/Work/Dev/GitHub/CristiFati/comtypes/src]> python -c "import comtypes"
Traceback (most recent call last):
  File "/mnt/e/Work/Dev/GitHub/CristiFati/comtypes/src/comtypes/__init__.py", line 8, in <module>
    from _ctypes import COMError
ImportError: cannot import name 'COMError' from '_ctypes' (/usr/lib/python3.8/lib-dynload/_ctypes.cpython-38-x86_64-linux-gnu.so)

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/mnt/e/Work/Dev/GitHub/CristiFati/comtypes/src/comtypes/__init__.py", line 16, in <module>
    raise ImportError(msg) from e
ImportError:
COM technology not available (maybe it's the wrong platform).
Note that COM is only supported on Windows.
For more details, please check: https://learn.microsoft.com/en-us/windows/win32/com
```
```
[cfati@cfati-5510-0:/mnt/e/Work/Dev/GitHub/CristiFati/comtypes/src]> python -c "import comtypes"
Traceback (most recent call last):
  File "/mnt/e/Work/Dev/GitHub/CristiFati/comtypes/src/comtypes/__init__.py", line 8, in <module>
    from _ctypes import COMError
ImportError: cannot import name 'COMError' from '_ctypes' (/usr/lib/python3.8/lib-dynload/_ctypes.cpython-38-x86_64-linux-gnu.so)

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/mnt/e/Work/Dev/GitHub/CristiFati/comtypes/src/comtypes/__init__.py", line 15, in <module>
    raise ImportError(msg) from e
ImportError: COM technology not available (maybe it's the wrong platform).
Note that COM is only supported on Windows.
For more details, please check: https://learn.microsoft.com/en-us/windows/win32/com
```

**Review by junkmd — CHANGES_REQUESTED** [review]
Such as `ImportError: COM technology...`, it would be easier to understand if there is a summary of the information beside it rather than under it.
Please remove the first line break.

**Comment by junkmd** [comment]
CI is failed…
I will close and reopen this to re-run appveyor.

**Comment by junkmd** [comment]
CIs are passed.

<details>
<summary>memo</summary>

https://ci.appveyor.com/project/cfarrow/comtypes/builds/45775836
https://ci.appveyor.com/project/pywinauto/pywinauto/builds/45775876
</details>

**Review by junkmd — APPROVED** [review]
Error message for using on non-Windows environments is improved. Thanks @CristiFati

**Pull request merged**